sky/packages/sky/lib/widgets/switch.dart

```diff
@@ -20,52 +25,124 @@ const double _kThumbRadius = 10.0;
 const double _kSwitchHeight = _kThumbRadius * 2.0;
 const double _kTrackHeight = 14.0;
 const double _kTrackRadius = _kTrackHeight / 2.0;
-const double _kTrackWidth = _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const double _kTrackWidth =
+    _kSwitchWidth - (_kThumbRadius - _kTrackRadius) * 2.0;
+const Duration _kCheckDuration = const Duration(milliseconds: 200);
+
+class Switch extends Component {
+  Switch({Key key, this.value, this.onChanged}) : super(key: key);
+
+  bool value;
+  ValueChanged onChanged;
+
+  Widget build() {
+    return new SwitchWrapper(
+        value: value,
+        onChanged: onChanged,
+        thumbColor: Theme.of(this).accentColor);
+  }
+}
 
-class Switch extends Toggleable {
-  // TODO(jackson): Hit-test the switch so that it can respond to both taps and swipe gestures
+// This wrapper class exists only because Switch needs to be a Component in
+// order to get an accent color from a Theme but Components do not know how to
+// host RenderObjects.
+class SwitchWrapper extends OneChildRenderObjectWrapper {
+  SwitchWrapper({Key key, this.value, this.onChanged, this.thumbColor})
```

Comment:
fyi the 'Format' functionality in my Atom removes these. i'm not sure where this formatting logic is coming from